src/Router.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Saritasa\LaravelControllers;
+
+use Illuminate\Container\Container;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Contracts\Routing\UrlRoutable;
+use Illuminate\Routing\Router as LaravelRouter;
+use Illuminate\Support\Str;
+use Illuminate\Routing\Route;
+use ReflectionParameter;
+use Saritasa\LaravelRepositories\Contracts\IRepositoryFactory;
+use Saritasa\LaravelRepositories\Exceptions\ModelNotFoundException;
+use Saritasa\LaravelRepositories\Exceptions\RepositoryException;
+
+/**
+ * Custom application router which use route mapping to resolve model binding.
+ */
+class Router extends LaravelRouter
+{
+    protected const ROUTE_MAPPING_KEY = 'mapping';
+
+    /**
+     * Repositories storage.
```

Comment:
not *storage*. But also no need to write that it is just *Repositories factory* because such description i can get by splitting variable name )